### PR TITLE
Add screentips to hat stabilizer component

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request
Initially this was added in:

- #87285

And then removed in:

- #87305

Not sure if @SmArtKar was aware that screentips can be added to components.

## Why It's Good For The Game
Consistency.

## Changelog
:cl:
fix: Fix missing screentips plasmaman helmets and MOD suit hat stabilizer helmets.
/:cl:

### CODE REVIEW

File: code/datums/components/hat_stabilizer.dm [hunk 1]
```diff
@@ -146,3 +151,17 @@
 	if (ismob(apparel.loc))
 		var/mob/wearer = apparel.loc
 		wearer.update_clothing(wearer.get_slot_by_item(apparel))
+
+// maybe we don't need the item context proc but instead the hand one? since we don't need to check held_item
+/datum/component/hat_stabilizer/proc/on_requesting_context_from_item(atom/source, list/context, obj/item/held_item, mob/user)
+	SIGNAL_HANDLER
+
+	if(attached_hat)
```
Comment (MrMelbert) [inline code/datums/components/hat_stabilizer.dm:159]:
```suggestion
	if(attached_hat && !held_item)
```

File: code/datums/components/hat_stabilizer.dm [hunk 2]
```diff
@@ -146,3 +151,17 @@
 	if (ismob(apparel.loc))
 		var/mob/wearer = apparel.loc
 		wearer.update_clothing(wearer.get_slot_by_item(apparel))
+
+// maybe we don't need the item context proc but instead the hand one? since we don't need to check held_item
```
Comment (MrMelbert) [inline code/datums/components/hat_stabilizer.dm:155]: This comment doesn't make any sense in this context, it says we don't check held item but we... do?

Comment (timothymtorres) [reply code/datums/components/hat_stabilizer.dm:155]: Sloppiness by me. I ported over this code from my rotation component refactor and forgot to clean it up.

Comment (MrMelbert) [reply code/datums/components/hat_stabilizer.dm]: Classic, I do it all the time

**Pull request merged**